### PR TITLE
Be/dictionaries

### PR DESCRIPTION
**Pull request by beccaelenzil**
Changes to address this feedback: Compare uniform dictionary use (e.g. frequency table) versus heterogeneous usage (more like an object)
https://app.asana.com/0/1201700438643002/1202377160745365/f

- Add section comparing heterogenous and homogenous dictionaries
- Mentioned that we will learn about objects (heterogenous)
- Mentioned benefit of homogenous is uniform processing

Would love feedback on clarity of explanation.

### CODE REVIEW

File: intro-to-python/dictionaries.md [hunk 1]
```diff
@@ -63,6 +63,52 @@ distance = planet_dict["venus"]
...
+
+```Python
+planet_dict = {
+    "mercury": "36 million miles", 
+    "venus": "67 million miles",
+    "mars": "142 million miles", 
+}
+```
+
+Notice that `planet_dict` is a *uniform* or *homogenous* dictionary. All the values have the same datatype, they are *strings*. Given this, we may chose to drop the `"million miles"` from the values and use *integers* for the values.
```
Comment (kp483) [inline intro-to-python/dictionaries.md:100]:
```suggestion
Notice that `planet_dict` is a *uniform* or *homogeneous* dictionary. All the values have the same data type: they are all *strings*. Given this, we may chose to drop the `"million miles"` from the values and instead use *integers* for the values. 
```

File: intro-to-python/dictionaries.md [hunk 2]
```diff
@@ -63,6 +63,52 @@ distance = planet_dict["venus"]
...
+    "street": ["315 5th Ave S", "Suite 200"]
+    "city": "Seattle"
+    "state": "WA"
+    "zip_code": 98104
+}
+```
+
+Notice that `adas_address` is a *heterogenous* dictionary. The values paired with `"name"`, `"city"`, and `"state"` are *strings*. The `"street"` value is a *list* holding the first and second lines of the address, and the `"zip_code"` is an *integer*.
+
+Later in the core curriculum, we will learn about classes and objects and how they are used to group and store data with different datatypes similar to a *heterogenous* dictionary.
```
Comment (kp483) [inline intro-to-python/dictionaries.md:86]:
```suggestion
*Heterogeneous* dictionaries are often used to group data with different data types together. Later in the core curriculum, we will learn about classes and objects and how they are similarly used to group and store data with different data types.
```

**Review by kp483 — APPROVED** [review]
A few nitpicky grammar and wording suggestions to try and put more emphasis on the differences between the two types of dictionaries and on the changes you wanted to highlight. Overall, looks good!

File: intro-to-python/dictionaries.md [hunk 3]
```diff
@@ -63,6 +63,52 @@ distance = planet_dict["venus"]
 # distance = "67 million miles"
 
 ```  
+
+## Heterogeneous vs. Homogeneous Dictionaries
+
+Dictionaries store data using key-value pairs. Let's look at a few examples where the datatype of the values is different within a single dictionary (heterogeneous) and where the datatype of the values is the same withing a single dictionary (homogeneous).
```
Comment (paulinah27) [inline intro-to-python/dictionaries.md:69]: Just a typo withing - within

Dictionaries store data using key-value pairs. Let's look at a few examples where the datatype of the values is different within a single dictionary (heterogeneous) and where the datatype of the values is the same within a single dictionary (homogeneous)

**Pull request merged**